scripts/pylib/twister/harness.py

```diff
@@ -121,6 +125,81 @@ def handle(self, line):
         else:
             self.tests[self.id] = "FAIL"
 
+class Pytest(Harness):
+
+    def configure(self, instance):
+        super(Pytest, self).configure(instance)
+        self.running_dir = instance.build_dir
+        self.source_dir = instance.testcase.source_dir
+        self.pytest_root = 'pytest'
+        config = instance.testcase.harness_config
+
+        if config:
+            self.pytest_root = config.get('pytest_root', 'pytest')
+
+    def handle(self, line):
+        ''' Test cases that make use of pytest more care about results given
+            by pytest tool which is called in pytest_run(), so works of this
+            handle is trying to give a PASS or FAIL to avoid timeout
+        '''
+        self.process_test(line)
+
+        if self.state == "passed":
+            self.tests[self.id] = "PASS"
+        else:
+            self.tests[self.id] = "FAIL"
```

Comment:
I think it is better to write the pytest harness in a way that no stub for output line processing is needed. When I debug this example each output line of the hello_world is still "checked" by this handle. `self.state` is none (not passed), hence  `self.tests[self.id] = "FAIL"` is executed. I think this can be confusing for anyone trying to debug the test process. Will it be possible to change the twister behavior at a higher level, i.e. when the harness is Pytest the `_output_handler` method wouldn't be needed / won't cause each output line to be checked?

Have you tried running this test with `-v -v`? It will spam the console with 
```
...
DEBUG   - OUTPUT: 
DEBUG   - OUTPUT: 
DEBUG   - OUTPUT: 
DEBUG   - OUTPUT: 
...
``` 
I think it is related with the above issue as this printout comes from `_output_handler` method,.

Reply:
Thank you @PerMac , I never use "-v -v" options so I didn't realize this problem. Now I see it.
So I check harness.is_pytest before calling harness.hanlde()  to avoid each output line to be checked.